app/models/spree/product.rb

```diff
@@ -474,5 +475,12 @@ def sanitize_permalink
       requested = permalink.presence || permalink_was.presence || name.presence || 'product'
       self.permalink = create_unique_permalink(requested.parameterize)
     end
+
+    def validate_image_for_master
+      return unless master.images.any? { |image| !image.valid? }
+
+      errors.add(:base, I18n.t('spree.admin.products.image_not_processable'))
+      false
```

Comment:
[Custom validation methods](https://guides.rubyonrails.org/active_record_validations.html#custom-methods) actually don't need to return anything; it's only the `errors.add` that matters.

```suggestion
```

Reply:
Thanks for the review, I have made the above changes.